pallets/committee/src/lib.rs

```diff
@@ -402,31 +439,28 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
             let seats = Self::members().len() as MemberCount;
             let yes_votes = voting.ayes.len() as MemberCount;
             let no_votes = voting.nays.len() as MemberCount;
+            let did = Context::current_identity::<Identity<T>>().unwrap_or_default();
+            Self::check_threshold_finalize(proposal, did, voting, yes_votes, no_votes, seats);
+        }
+    }
 
-            let threshold = <VoteThreshold<I>>::get();
-
-            let approved = Self::is_threshold_satisfied(yes_votes, seats, threshold);
-            let rejected = Self::is_threshold_satisfied(no_votes, seats, threshold);
-
-            if approved || rejected {
-                let current_did = Context::current_identity::<Identity<T>>().unwrap_or_default();
-                Self::finalize_proposal(
-                    approved,
-                    seats,
-                    yes_votes,
-                    no_votes,
-                    proposal,
-                    current_did,
-                );
-                Self::deposit_event(RawEvent::FinalVotes(
-                    current_did,
-                    voting.index,
-                    proposal,
-                    voting.ayes,
-                    voting.nays,
-                ));
-            }
+    fn check_threshold_finalize(
+        proposal: T::Hash,
+        did: IdentityId,
+        voting: PolymeshVotes<IdentityId, T::BlockNumber>,
+        yes_votes: MemberCount,
+        no_votes: MemberCount,
+        seats: MemberCount,
+    ) {
+        let threshold = <VoteThreshold<I>>::get();
+        let approved = Self::is_threshold_satisfied(yes_votes, seats, threshold);
+        let rejected = Self::is_threshold_satisfied(no_votes, seats, threshold);
```

Comment:
Minor optimization: do the calculation only for the larger of approved/rejected.